wikimedia/wikipedia-ios
# Change payment method prompt from alert view to action sheet

**Phabricator:** 
https://phabricator.wikimedia.org/T345851#9236850

### Notes
This PR changes the payment method prompt from alert view to action sheet.

Prior to test steps: make [these changes](https://github.com/wikimedia/wikipedia-ios/pull/4655/commits/31b5386123345f5f52ffa4cf59defc1672578210) from #4655 locally, if needed.

### Test Steps
1. Log into iOS Sandbox Tester Apple ID (see 1Password) in test device iOS Settings. Do not enable 2 factor auth.
In Wallet app, add [Test Apple Pay card](https://developer.apple.com/apple-pay/sandbox-testing/).
2. Change device region to NL or IT.
3. On Staging scheme, fresh install and launch app.
4. On Explore feed, background, then foreground, then pull to refresh on Explore feed to fetch latest campaigns.
5. In Settings, you should now see an  action sheet after tapping "Donate with Apple Pay" instead of an alert view.
6. Visit an article on EN Wikipedia. Tap "Donate" on the campaign modal. You should now see an action sheet instead of an alert view.

### Screenshots/Videos
![IMG_0242](https://github.com/wikimedia/wikipedia-ios/assets/3620196/af1e4e02-6fbe-41f0-9264-b4bfe7395307)
![IMG_0241](https://github.com/wikimedia/wikipedia-ios/assets/3620196/dc4d8d00-60d1-4860-be37-276869b054f2)

![IMG_1271](https://github.com/wikimedia/wikipedia-ios/assets/3620196/75f3497c-2cbd-49dd-955f-80fe99bd66ef)
![IMG_1266](https://github.com/wikimedia/wikipedia-ios/assets/3620196/b23986bc-6020-453a-a4fd-7755efdf5c51)
![IMG_1270](https://github.com/wikimedia/wikipedia-ios/assets/3620196/546e1ef2-33a5-4d2a-bb95-38b5b660890e)
![IMG_1269](https://github.com/wikimedia/wikipedia-ios/assets/3620196/d3a3bb00-d933-403d-b904-93122085f1af)
